Source/JavaScriptCore/runtime/AtomicsObject.cpp

```diff
@@ -498,6 +563,37 @@ JSC_DEFINE_HOST_FUNCTION(atomicsFuncWait, (JSGlobalObject* globalObject, CallFra
     return { };
 }
 
+EncodedJSValue getWaiterListSize(JSGlobalObject* globalObject, CallFrame* callFrame)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* typedArrayView = validateIntegerTypedArray<TypedArrayOperationMode::Wait>(globalObject, callFrame->argument(0));
+    RETURN_IF_EXCEPTION(scope, { });
+
+    if (!typedArrayView->isShared())
+        throwVMTypeError(globalObject, scope, "Typed array for wait/waitAsync/notify must wrap a SharedArrayBuffer."_s);
```

Comment:
Should it be `return throwVMTypeError` here? If that's the case, it would be nice to add a test that captures that (like incorrect / throwing `callFrame->argument(1)` doesn't overwrite the `"Typed array for wait/waitAsync/notify must wrap a SharedArrayBuffer."` error).